phalcon/html/helper/abstracthelper.zep

```diff
@@ -0,0 +1,134 @@
+
+/**
+ * This file is part of the Phalcon Framework.
+ *
+ * (c) Phalcon Team <team@phalconphp.com>
+ *
+ * For the full copyright and license information, please view the LICENSE.txt
+ * file that was distributed with this source code.
+ */
+
+namespace Phalcon\Html\Helper;
+
+use Phalcon\Html\Exception;
+use Phalcon\EscaperInterface;
+
+/**
+ * Phalcon\Html\Helper\AbstractHelper
+ *
+ * Abstract class for all html helpers
+ */
+abstract class AbstractHelper
```

Comment:
This class should implement an interface with the __invoke method

Reply:
We can't put `__invoke()` there because it is not the same `__invoke` for every helper. For instance one might get instantiated with 2 parameters another with 3. See `Anchor` (3) vs `Button` (2)

Reply:
Best practice is to use an interface to specify the _contract_ and an abstract class as just one _implementation_ thereof. That abstract class can fill in a lot of the boilerplate so you can create an implementation by just overriding what you need to or want to without forcing you to use a particular implementation.

Reply:
How can you override the parameters of a method (number or type)? There is no interface that I can abstract from the helpers since they all call `__invoke` with different parameters based on the type. The abstract helper class is there to offer common functionality for the helpers.We can easily rename it to `Base` or something if Abstract is not ok.

Reply:
Well, if the design implies that the methods may have different signatures and this cannot be circumvented, then I cannot argue